.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,42 @@
+name: golangci-lint
+
+on:
+  push:
+    branches:
+      - develop
+      - main
+    paths:
+      - '**.go'
```

Comment:
Will this only run if '**.go' files are touched?

If that's the case, then we can't enable this as a global check for develop branch. Consider removing this filter, for now, so we enable this check globally.